lib/pure/oids.nim

```diff
@@ -66,35 +67,33 @@ proc oidToString*(oid: Oid, str: cstring) =
   str[24] = '\0'
 
 proc `$`*(oid: Oid): string =
+  ## Converts an oid to string.
   result = newString(24)
   oidToString(oid, result)
 
-proc rand(): cint {.importc: "rand", header: "<stdlib.h>", nodecl.}
-proc srand(seed: cint) {.importc: "srand", header: "<stdlib.h>", nodecl.}
-
-var t = getTime().toUnix.int32
-srand(t)
 
 var
-  incr: int = rand()
-  fuzz: int32 = rand()
+  t = getTime().toUnix.int32
+  seed = initRand(t)
+  incr: int = seed.rand(int.high)
+
+let fuzz = cast[int32](seed.rand(high(int)))
 
 proc genOid*(): Oid =
-  ## generates a new OID.
+  ## Generates a new OID.
+  runnableExamples:
+    doAssert ($genOid()).len == 24
+    if false: doAssert $genOid() == "5fc7f546ddbbc84800006aaf"
   t = getTime().toUnix.int32
-  var i = int32(atomicInc(incr))
+  var i = cast[int32](atomicInc(incr))
```

Comment:
`atomicInc` is meaningless in this context, also why `incr` is an `int` that's downcasted to int32?

Reply:
https://github.com/nim-lang/Nim/pull/16203#discussion_r537868831

Reply:
Seems to me rand(low..high(int32)) would have the same effect

Reply:
> atomicInc is meaningless in this context

`atomicInc` is for thread safety.

> why incr is an int that's downcasted to int32?

inc is to follow oid standard (for consecutive oids generated); cast is to avoid overflow errors; int32 because... oid standard; not sure what else you're suggesting here.

> Seems to me rand(low..high(int32)) would have the same effect

I don't see how that would follow oid standard; at least this needs much more details in your explanation